core/src/main/java/org/opensearch/sql/expression/operator/convert/TypeCastOperator.java

```diff
@@ -60,6 +62,7 @@ public static void register(BuiltinFunctionRepository repository) {
     repository.register(castToTime());
     repository.register(castToTimestamp());
     repository.register(castToDatetime());
+    repository.register(typeof());
```

Comment:
Could you please move `typeof` into a separate class?

Reply:
How can I call it? Is `Misc` OK?

Reply:
`TypeOfOperator` seems right to me.

Reply:
Fixed in 9296c621.